Configure

```diff
@@ -1830,6 +1838,14 @@ $config{CFLAGS} = [ map { $_ eq '--ossl-strict-warnings'
                               : ( $_ ) }
                     @{$config{CFLAGS}} ];
 
+if ($disable_engine_removal == 1) {
+    disable("engine-removal", "engine");
+    disable("engine-removal", "afalgeng");
+    disable("engine-removal", "capieng");
+    disable("engine-removal", "loadereng");
+    disable("engine-removal", "padlockeng");
+}
+
```

Comment:
I'm not 100% sure about this approach.  It makes sense what you're trying to do, but if I'm reading it correctly, this effectively overrides whatever configuration is passed to disable engines.

Thats certainly efficient, but I was under the impression that what we wanted was effectively a TODO marker in each test that was failing (or will fail, or be skipped) as a result of not having engine support.  That way we can, at the end of the development of this removal effort , we can audit the code and ensure each test that we marked as such is either removed (as it is no longer needed), or properly updated to use a provider instead.

Or did I misunderstand something?

Reply:
>  if I'm reading it correctly, this effectively overrides whatever configuration is passed to disable engines.
Yes, that's correct. The plan was to manually disable all the tests affected, but I noticed in most of the cases that possibility is already there, so I don't see the reason to reinvent the wheel. To outcome overriding the config options (which are going away anyway) or skipping them manually is going to have the same output: regardless of the configuration, the tests won't run. I find this a lot less complex approach and the tests can be whenever easily re-enabled.
If you wish, I can add each test a separate skip reason and mark them with some TODO label with the same kind of option to disable them in the config script, but it'll be just extra work to do

Reply:
I do  not see all details (yet :), but I actually like this simple solution.

Hovewer, as said above, all engine tests must be either removed or ported to providers. We should not forget anything in the disabled state anywhere, so this hack must be completely removed too. I think this can be verified in the final merge review, though.

Reply:
Actually this also solves the issue I had in the first PR (https://github.com/openssl/openssl/pull/28384) where fcntl symbol got imported for "no reason". The reason was that, when the engines where not disabled, some of the engine directory code got pulled in and thrashed the build. This PR resolves that issue (I tried to apply this patch there and the tests run successfully: https://github.com/openssl/openssl/actions/runs/17488723258/job/49673450115?pr=28384)

The engines should go from the build config sooner or later anyway, not talking about the random build breakages I had above

Reply:
ok, then I'm ok with this approach, but I would appreciate (if you haven't done so already), issues to track each test case that that is failing with this not in place, so we can easily see what tests we need to address in our issue list.

Reply:
issues were created